distributed/client.py

```diff
@@ -1310,8 +1310,9 @@ async def _close(self, fast=False):
 
         self.status = "closing"
 
-        for pc in self._periodic_callbacks.values():
-            pc.stop()
+        with suppress(AttributeError):
+            for pc in self._periodic_callbacks.values():
+                pc.stop()
```

Comment:
Why was this change needed?

Reply:
That's a good question.  I'm getting tracebacks like the following, which I don't really understand.  I guess Python can delete an object while it's initializing?  Or maybe this is an at shutdown time thing.

```python
distributed.http.proxy - INFO - To route to workers diagnostics web server please install jupyter-server-proxy: python -m pip install jupyter-server-proxy
distributed.scheduler - INFO - Clear task state
distributed.scheduler - INFO -   Scheduler at:     tcp://127.0.0.1:41303
distributed.scheduler - INFO -   dashboard at:           127.0.0.1:36750
Exception ignored in: <function Client.__del__ at 0x7f2d672f2440>
Traceback (most recent call last):
  File "/home/travis/build/dask/distributed/distributed/client.py", line 1189, in __del__
    self.close()
  File "/home/travis/build/dask/distributed/distributed/client.py", line 1413, in close
    if self.asynchronous:
  File "/home/travis/build/dask/distributed/distributed/client.py", line 802, in asynchronous
    return self._asynchronous and self.loop is IOLoop.current()
AttributeError: 'Client' object has no attribute '_asynchronous'
```

Reply:
But they're probably not needed.  I can remove them.  I'm just trying a few things to try to placate the stress tests.

Reply:
> I guess Python can delete an object while it's initializing?

Once `__new__` is called and the object instance is created, the `__del__` is always called. So if an error occurs in `__init__` and initialization fails the `__del__` will still be called.